src/itype.cpp

```diff
@@ -4,14 +4,25 @@
 #include "game.h"
 #include "item_factory.h"
 #include "translations.h"
+#include "output.h"
 
 #include <fstream>
 #include <stdexcept>
 #include <algorithm>
 
 std::string itype::nname( unsigned int const quantity ) const
 {
-    return ngettext( name.c_str(), name_plural.c_str(), quantity );
+    std::string str( ngettext( name.c_str(), name_plural.c_str(), quantity ) );
+
+    if( wheel ) {
+        return string_format( str, wheel->diameter );
+
+    } else if( engine && engine->displacement > 0 ) {
+        return string_format( str, engine->displacement / 100.0 );
```

Comment:
A single innocent '%' in the name invokes undefined behavior (format specifiers not matching the arguments). The feature is also not documented at all.

You should use a separate format string, like it's done for the morale description ("Enjoyed %i"), and replace that manually - there is a `replace_substring` function in output.h.

> It's not that you don't have a point just that the problem gets pushed elsewhere by any different solution and increasing the difficulty of i18n costs us both developers and players.

So does undefined behavior.